internal/checker/terraform/rules/gcp/google_iam_folder_dataproc_private_cluster.rego

```diff
@@ -0,0 +1,76 @@
+# METADATA
+# title: "Verify Dataproc Clusters Restrict Public Access"
+# description: "This policy ensures that Dataproc clusters are configured to restrict public access, preventing unauthorized exposure to the public internet."
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataproc_cluster_iam
+# custom:
+#   id: CB_TFGCP_108
+#   severity: HIGH
+package lib.terraform.CB_TFGCP_108
+
+import future.keywords.in
+
+isvalid(block) {
+	supported_resources := ["google_dataproc_cluster_iam_member", "google_dataproc_cluster_iam_binding"]
+	block.Type == "resource"
+	some label in block.Labels
+	label in supported_resources
+}
+
+resource[resource] {
+	some block in pass
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	some block in fail
+	resource := concat(".", block.Labels)
+}
+
+pass[resource] {
+	some resource in input
+	isvalid(resource)
+	not fail[resource]
+}
+
+fail[resource] {
+	some resource in input
+	isvalid(resource)
+	resource.Type == "resource"
+	"google_dataproc_cluster_iam_member" in resource.Labels
+	invalid_members := [
+		"allUsers",
+		"allAuthenticatedUsers",
+	]
+	some member in resource.Attributes.member
```

Comment:
attribute member in "google_dataproc_cluster_iam_member" is not an array this should be 

resource.Attributes.member in invalid_members

and try to declare invalid members outside the block it has been used in line 49-59

Reply:
Thank you for noticing, noted on this one.